sdk/keyvault/keyvault-certificates/src/identifier.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { parseKeyvaultIdentifier } from "./core/utils";
+
+/**
+ * Represents a KeyVault identifier and its parsed contents.
+ */
+export interface ParsedKeyVaultCertificatesIdentifier {
+  /**
+   * The type of resource under KeyVault that this identifier is referring to.
+   */
+  collection: "certificates";
+
+  /**
+   * The originally received identifier.
+   */
+  id: string;
+
+  /**
+   * The KeyVault Key unique identifier (an URl).
+   */
+  vaultUrl: string;
+
+  /**
+   * The version of KeyVault Key. May be undefined.
+   */
+  version?: string;
+
+  /**
+   * The name of the KeyVault Key.
+   */
+  name: string;
+}
+
+/**
+ * Parser of the KeyVaultIdentifier for the KeyVault Keys Client.
+ */
+export class KeyVaultCertificatesIdentifier implements ParsedKeyVaultCertificatesIdentifier {
+  /**
+   * The type of resource under KeyVault that this identifier is referring to.
+   */
+  collection: "certificates";
+
+  /**
+   * The originally received identifier.
+   */
+  public id: string;
+
+  /**
+   * The KeyVault Key unique identifier (an URl).
+   */
+  public vaultUrl: string;
+
+  /**
+   * The version of KeyVault Key. May be undefined.
+   */
+  public version?: string;
+
+  /**
+   * The name of the KeyVault Key.
+   */
+  public name: string;
+
+  constructor(url: string) {
+    const collection = "certificates";
+    const coreParsedIdentifier = parseKeyvaultIdentifier(collection, url);
```

Comment:
What about deleted certs, keys, and secrets? Someone might want to pass, for example, a `DeletedCertificate.Id`. Then the question becomes: do we still say the collection is, in this case, "certificates" or "deletedcertificates"?

Reply:
I'm pretty sure it's certificates in that case, based on the generated code.

Reply:
Here's an ID of a certificate I just deleted and retrieved: `https://danrodri-kv-2020-1.vault.azure.net/certificates/recoverCertificateName-cangetadeletedcertificate-15203725962992753/26d7cc3645b146fe80d7bb56e7aba0d6`